lib/pbench/server/cache_manager.py

```diff
@@ -274,59 +277,51 @@ def cache_map(self, tar_path: Path):
             for l_path in dir_list:
                 tar_info = FileInfo(l_path)
                 curr[l_path.name] = {"details": tar_info}
-                # Note: have to handle symlink case as it
-                # is not being checked whether it is a symlink,
-                # it is checking for the redirected path. Currently,
-                # as I have used file for the symlink it is not added it
-                # the queue, but if it a symlink is pointing to a directory
-                # it turns out to be an endless loop. There may be case
-                # that the way I am creating symlink in tests is wrong because
-                # when I am testing manually I am not seeing the same case.
                 if l_path.is_dir():
                     dir_queue.append((l_path, curr))
         self.cachemap = cmap
 
     @staticmethod
-    def traverse_cmap(path, cachemap):
+    def traverse_cmap(path: Path, cachemap: dict):
```

Comment:
Missing return type hint.

Reply:
> I think we need to handle errors if, at any point in the traversal, we don't find the name we're expecting, or if a non-leaf reference isn't a directory...

Yeah...I presume that we're going to provide a Pbench Server API which accepts a path and returns the corresponding item from the cache (either its metadata or its contents, depending), in which case we're going to have to defend against bad paths.